### DISCUSSION TITLE
support UDP

### DISCUSSION THREAD

**Discussion by ArtemIsmagilov** [memcached/memcached#1158] [General]
Hello, please tell me, I want to add support for a client using the UDP protocol. This should improve performance. I understood this protocol a little but not completely. You also have the UDP protocol disabled by default for this, which is great. But I don't understand. 1.What should I be careful about when adding UDP support to my client? In public sources there is only information about DDOS attacks and this is alarming. 
2.Why doesn't redis deliberately support UDP to improve performance?
I am aware that many servers and utilities run on UDP. 
3. Or is this protocol supported due to compatibility reasons with older versions of memcached?

**Comment by ArtemIsmagilov** [comment]
https://github.com/memcached/memcached/wiki/ConfiguringServer#networking

**Comment by dormando** [comment]
It doesn't actually improve performance more than it increases the complexity. If it were free performance redis would support it.

In short you end up re-inventing TCP anyway, so let TCP do the right thing. You should also read those articles on the DDoS if you haven't.

**Reply by ArtemIsmagilov** [reply]
Thank you very much for answering)
